website/docs/module-type-overrides.md

```diff
@@ -41,6 +41,29 @@ When multiple patterns match the same file, the last pattern takes precedence.
 * `esm` overrides matches files to compile and execute as native ECMAScript modules.
 * `package` resets either of the above to default behavior, which obeys `package.json` `"type"` and `tsconfig.json` `"module"` options.
 
+**Note:** This does not mean that you have to write CommonJS in your source `webpack.config.ts` file
+- here you can continue writing imports and exports in ESM syntax.
+
+For example, this syntax with `import` and `export` in your `webpack.config.ts` source file is just fine:
+
+```ts
+import * as path from 'path';
+import * as webpack from 'webpack';
+// in case you run into any typescript error when configuring `devServer`
+import 'webpack-dev-server';
+
+const config: webpack.Configuration = {
+  mode: 'production',
+  entry: './foo.js',
+  output: {
+    path: path.resolve(__dirname, 'dist'),
+    filename: 'foo.bundle.js',
+  },
+};
+
+export default config;
+```
```

Comment:
I hesitate to duplicate this rather than linking to webpack's documentation.  webpack might be updating it soon.  Also, webpack is not the only tool we want to integrate with.

Reply:
How about we create a webpack page under "Recipes"?  That way, we can afford to give very specific webpack advice and examples, and we can link to the moduleTypes page.

Reply:
Well, if we just follow your other suggestion from above and link to the webpack config, would that resolve this in a different way?

Reply:
I guess the only thing we'd be saying that webpack's website isn't, is we'd be giving an example `moduleTypes` config.  I dunno if webpack's website will eventually add the same.

Reply:
Right, I think I understand - you mean because if it was a recipe, both the `moduleTypes` config and the actual webpack config would just be in one place, got it. That's true - if we linked to the docs instead, then the `moduleTypes` config would be on the `ts-node` website and the webpack config would be on the webpack docs website, separated by that link.